### ISSUE TITLE
[Feature Request] Chimera's Breath shader

### ISSUE THREAD

**Issue by Anonymous4045** [y4my4my4m/kde-shader-wallpaper#41]
With shaders being able to react to the mouse now, like in CubeLines, one shader I think would be really cool is fluid simulation that interacts with the cursor. One example on ShaderToys is [Chimera's Breath](https://www.shadertoy.com/view/4tGfDW). I tried compiling the source with qsb, but I'm not sure how any of this works and ran into many errors.

**Comment by y4my4my4m** [comment]
@Anonymous4045 
Indeed, that one and this one https://www.shadertoy.com/view/Xsd3DB are some shaders I hope to be able to bring to kde-shader-wallpaper.

The reason for the compilation not work is because those shaders uses buffers, which is not yet supported.

Ultimately i'd really like shaders to be able to interact with the windows too (imagine you move a window around and it creates smoke/water effect around it).

**Comment by Anonymous4045** [comment]
@y4my4my4m Ah, shame. Do you know if the devs plan to add buffer support anytime soon?

**Comment by y4my4my4m** [comment]
@Anonymous4045 
So essentially, I'm still not sure it's impossible to do.
[I've been testing stuff](https://github.com/y4my4my4m/kde-shader-wallpaper/blob/0db87022d8578a2e21e3cc5626d6487a0a899d21/package/contents/ui/main.qml#L134-L163) with ShaderEffectSource and got *some* results...but I still don't fully understand.

If anything, I believe it should be possible to achieve were I to make a C++ wrapper for it.

**Comment by y4my4my4m** [comment]
This should now be possible on V4 since buffers are supported!
